### ISSUE TITLE
`experimental:spacing-between-declarations-with-annotations` false positive

### ISSUE THREAD

**Issue by BaloghTamas** [pinterest/ktlint#1218]
## Expected Behavior
```kotlin
/**
 * Some comment
 */
@JvmStatic
fun doSomething() {
    // ...
}
```

For the above code there should be no error reported.

## Observed Behavior
Error is reported:
```
 Lint error > [experimental:spacing-between-declarations-with-annotations] Declarations and declarations with annotations should have an empty space between.
```

## Steps to Reproduce
Described above.

## Your Environment
* Version of ktlint used: 0.42.1  (The issue was not present in 0.41.0)

**Comment by BaloghTamas** [comment]
Duplicated of https://github.com/pinterest/ktlint/issues/1053

**Comment by BaloghTamas** [comment]
Might be related to #1053

**Comment by paul-dingemans** [comment]
This has been fixed by https://github.com/pinterest/ktlint/pull/1291

**Comment by paul-dingemans** [comment]
> This has been fixed by #1291

Closing issue
